File: server/analyze_construction.py
import sqlite3
import json
from pathlib import Path

db_path = Path(__file__).parent / 'construction.db'

conn = sqlite3.connect(str(db_path))
conn.row_factory = sqlite3.Row
cursor = conn.cursor()

try:
    # 모든 테이블 조회
    cursor.execute("SELECT name FROM sqlite_master WHERE type='table' ORDER BY name;")
    tables = cursor.fetchall()
    
    print("=== 데이터베이스 테이블 ===")
    table_names = [row['name'] for row in tables]
    print(json.dumps(table_names, indent=2, ensure_ascii=False))
    
    # 각 테이블의 스키마 조회
    print("\n=== 테이블 스키마 ===")
    for table_name in table_names:
        cursor.execute(f"PRAGMA table_info({table_name});")
        schema = cursor.fetchall()
        print(f"\n[{table_name}]")
        for col in schema:
            print(f"  {col['name']}: {col['type']}")
    
    # buildings 테이블 데이터
    if 'buildings' in table_names:
        print("\n=== buildings 테이블 데이터 ===")
        cursor.execute("SELECT * FROM buildings;")
        buildings = cursor.fetchall()
        for row in buildings:
            print(dict(row))
    
    # sites 테이블 데이터
    if 'sites' in table_names:
        print("\n=== sites 테이블 데이터 ===")
        cursor.execute("SELECT * FROM sites;")
        sites = cursor.fetchall()
        for row in sites:
            print(dict(row))
    
    # units 테이블 데이터 (처음 50개)
    if 'units' in table_names:
        print("\n=== units 테이블 데이터 (처음 50개) ===")
        cursor.execute("SELECT * FROM units LIMIT 50;")
        units = cursor.fetchall()
        for row in units:
            print(dict(row))
        
        cursor.execute("SELECT COUNT(*) as count FROM units;")
        count = cursor.fetchone()
        print(f"\n총 units 개수: {count['count']}")

except Exception as error:
    print(f'DB 조회 오류: {error}')

finally:
    conn.close()

File: server/analyze_construction.js
const Database = require('better-sqlite3');
const path = require('path');

// DB 파일 경로
const dbPath = path.join(__dirname, 'construction.db');
const db = new Database(dbPath);

try {
  // 모든 테이블 조회
  const tables = db.prepare(`
    SELECT name FROM sqlite_master 
    WHERE type='table' 
    ORDER BY name;
  `).all();

  console.log('=== 데이터베이스 테이블 ===');
  console.log(JSON.stringify(tables, null, 2));

  // 각 테이블의 스키마 조회
  console.log('\n=== 테이블 스키마 ===');
  tables.forEach(table => {
    const schema = db.prepare(`PRAGMA table_info(${table.name});`).all();
    console.log(`\n[${table.name}]`);
    console.log(JSON.stringify(schema, null, 2));
  });

  // buildings 테이블 데이터 확인
  if (tables.some(t => t.name === 'buildings')) {
    console.log('\n=== buildings 테이블 데이터 ===');
    const buildings = db.prepare('SELECT * FROM buildings;').all();
    console.log(JSON.stringify(buildings, null, 2));
  }

  // sites 테이블 데이터 확인
  if (tables.some(t => t.name === 'sites')) {
    console.log('\n=== sites 테이블 데이터 ===');
    const sites = db.prepare('SELECT * FROM sites;').all();
    console.log(JSON.stringify(sites, null, 2));
  }

  // units 테이블 데이터 확인
  if (tables.some(t => t.name === 'units')) {
    console.log('\n=== units 테이블 데이터 ===');
    const units = db.prepare('SELECT * FROM units LIMIT 50;').all();
    console.log(JSON.stringify(units, null, 2));
    const unitCount = db.prepare('SELECT COUNT(*) as count FROM units;').get();
    console.log(`\n총 units 개수: ${unitCount.count}`);
  }

} catch (error) {
  console.error('DB 조회 오류:', error);
} finally {
  db.close();
}
